Source/monster.h

```diff
@@ -167,107 +166,115 @@ struct CMonster {
 };
 
 struct Monster { // note: missing field _mAFNum
-	const char *mName;
-	CMonster *MType;
-	const MonsterData *MData;
-	std::unique_ptr<uint8_t[]> uniqueTRN;
-	AnimationInfo AnimInfo;
-	int _mgoalvar1;
-	int _mgoalvar2;
-	int _mgoalvar3;
-	int _mVar1;
-	int _mVar2;
-	int _mVar3;
-	int _mmaxhp;
-	int _mhitpoints;
-	uint32_t _mFlags;
+	const char *name;
+	CMonster *type;
+	const MonsterData *data;
+	std::unique_ptr<uint8_t[]> uniqueMonsterTRN;
+	AnimationInfo animInfo;
+	/** Specifies current goal of the monster */
+	monster_goal goal;
+	/** Specifies monster's behaviour regarding moving and changing goals. */
+	int goalGeneral;
+	/** @brief Specifies turning direction for @p RoundWalk in most cases.
+	 *  Used in custom way by @p FallenAi, @p SnakeAi, @p M_FallenFear and @p FallenAi.
+	 */
+	int goalTurning;
+	/** @brief Controls monster's behaviour regarding special actions.
+	 *  Used only by @p ScavengerAi and @p MegaAi.
+	 */
+	int goalSpecialAction;

```

Comment:
```suggestion
	int goalVar1;
	/**
	 * @brief Specifies turning direction for @p RoundWalk in most cases.
	 * Used in custom way by @p FallenAi, @p SnakeAi, @p M_FallenFear and @p FallenAi.
	 */
	int goalVar2;
	/**
	 * @brief Controls monster's behaviour regarding special actions. 
	 * Used only by @p ScavengerAi and @p MegaAi.
	 */
	int goalVar3;
```

This is also used for transferering an X coordinate in a X/Y pair with `goalTurning`, so best keep them as `goalVar1`,  `goalVar2`. `goalSpecialAction` is also not used enough for monsters special action that I don't think it warrants renaming when the others are still there.

Nice that you added a meaningful name and description to these :+1:

Reply:
Yes, those members are problematic: they are used widely in monsters' ai function, sometimes to do seemingly unrelated things (`goalGeneral`  and `goalTurning` being used to calculate X and Y at one point - as you have said). 

I see the reasons to leave them as `goalVars` but those names are enigmatic, at least to me. Still, my solution in this commit reflects only a part of their usage - not ideal.

I checked the code and I think that X & Y calculation are done only by ScavengerAi - the rest seem to use `goalGeneral` as the name and description suggests. Still problematic to come up with an ideal name...

`goalSpecialAction` is used only by two Ais so maybe the name should reflect that?

Reply:
Maybe revert these for now, then we can take them separately in a different PR as they are going to take some more debating before we will probably land on good meaningful names for them.